[DO NOT MERGE] Add flutter_soloud (audio) codelab

filiph opened the pull request:
This PR adds the code of the upcoming audio engine codelab (using `flutter_soloud`).

The code is ready for review but I'd wait with merging even if it does pass CI/CD and review on its first pass. For example, the sample depends on the `2.0.0-pre.4` preview version of `flutter_soloud` — which will soon be replaced with `2.0.0`.


## Pre-launch Checklist

- [x] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

<!-- Links -->
[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

domesticmouse commented:
First question, what platforms does `soloud` support? If any of the six aren't supported, we should selectively generate the runners for only the supported platforms. By the look of https://pub.dev/packages/flutter_soloud we should be dropping the `web` runner? This can be accomplished by tuning the `flutter create` arguments. The keyword to look for in the help listing is `platform`.

Second, please consider using `flutter create --empty` as the basis of the codelab. I find it's more stable a platform than the default generated template app.

filiph commented:
Addressed the feedback: using `--empty` and and excluding `web` as platform.

All tests pass, except one connection error flake in one of the other codelabs:

```
 == Getting dependencies for next-gen-ui/step_03_b
/d/a/codelabs/codelabs/next-gen-ui/step_03_b /d/a/codelabs/codelabs/next-gen-ui/step_03_b /d/a/codelabs/codelabs
Resolving dependencies...
Git error. Command: `git fetch`
stdout: 
stderr: fatal: unable to access 'https://github.com/google/flutter-desktop-embedding.git/': Failed to connect to github.com port 443 after 21066 ms: Couldn't connect to server
```

domesticmouse commented:
> All tests pass, except one connection error flake in one of the other codelabs

I have a brute-force fix for that in https://github.com/flutter/codelabs/pull/2179

domesticmouse commented:
This should now be landable. =)

filiph commented:
Alright! The last change is there, so feel free to merge this at any time.

Thanks for the patience!
